inference-engine/src/mkldnn_plugin/nodes/mkldnn_non_max_suppression_node.cpp

```diff
@@ -15,8 +15,552 @@
+        // x of intersection
+        uni_vminps(vmm_temp4, vmm_boxes_coord3, vmm_candidate_coord3);  // min(Xmax)
+        uni_vmaxps(vmm_temp2, vmm_boxes_coord1, vmm_candidate_coord1);  // max(Xmin)
+        uni_vsubps(vmm_temp4, vmm_temp4, vmm_temp2);  // min(Xmax) - max(Xmin)
+        uni_vmaxps(vmm_temp4, vmm_temp4, vmm_zero);
+
+        // intersection_area
+        uni_vmulps(vmm_temp3, vmm_temp3, vmm_temp4);
+
+        // iou: intersection_area / (areaI + areaJ - intersection_area);
+        uni_vsubps(vmm_temp1, vmm_temp1, vmm_temp3);
+        uni_vdivps(vmm_temp3, vmm_temp3, vmm_temp1);
+    }
+
+    // std::exp(scale * iou * iou)
+    inline void soft_coeff() {
+        uni_vmulps(vmm_temp3, vmm_temp3, vmm_temp3);
+        uni_vmulps(vmm_temp3, vmm_temp3, vmm_scale);
+        exp_injector->compute_vector_range(vmm_temp3.getIdx(), vmm_temp3.getIdx() + 1);
+    }
+
+    inline void horizontal_mul_xmm(const Xbyak::Xmm &xmm_weight, const Xbyak::Xmm &xmm_aux) {
+        uni_vmovshdup(xmm_aux, xmm_weight);              //  weight:1,2,3,4; aux:2,2,4,4
+        uni_vmulps(xmm_weight, xmm_weight, xmm_aux);     //  weight:1*2,2*2,3*4,4*4
+        uni_vmovhlps(xmm_aux, xmm_aux, xmm_weight);      //  aux:3*4,4*4,4,4
+        uni_vmulps(xmm_weight, xmm_weight, xmm_aux);     //  weight:1*2*3*4,...
+    }
+
+    // horizontal mul for vmm_weight(Vmm(3)), temp1 and temp2 as aux
+    inline void horizontal_mul() {
+        Xbyak::Xmm xmm_weight = Xbyak::Xmm(vmm_temp3.getIdx());
+        Xbyak::Xmm xmm_temp1 = Xbyak::Xmm(vmm_temp1.getIdx());
+        Xbyak::Xmm xmm_temp2 = Xbyak::Xmm(vmm_temp2.getIdx());
+        if (isa == cpu::x64::sse41) {
+            horizontal_mul_xmm(xmm_weight, xmm_temp1);
+        } else if (isa == cpu::x64::avx2) {
+            Xbyak::Ymm ymm_weight = Xbyak::Ymm(vmm_temp3.getIdx());
+            vextractf128(xmm_temp1, ymm_weight, 0);
+            vextractf128(xmm_temp2, ymm_weight, 1);
+            uni_vmulps(xmm_weight, xmm_temp1, xmm_temp2);
+            horizontal_mul_xmm(xmm_weight, xmm_temp1);
+        } else {
+            Xbyak::Zmm zmm_weight = Xbyak::Zmm(vmm_temp3.getIdx());
+            vextractf32x4(xmm_temp1, zmm_weight, 0);
+            vextractf32x4(xmm_temp2, zmm_weight, 1);
+            uni_vmulps(xmm_temp1, xmm_temp1, xmm_temp2);
+            vextractf32x4(xmm_temp2, zmm_weight, 2);
+            vextractf32x4(xmm_weight, zmm_weight, 3);
+            uni_vmulps(xmm_weight, xmm_weight, xmm_temp2);
+            uni_vmulps(xmm_weight, xmm_weight, xmm_temp1);
+            horizontal_mul_xmm(xmm_weight, xmm_temp1);
+        }
+    }
+
+    inline void prepare_table() {
+        auto broadcast_d = [&](int val) {
+            for (size_t d = 0; d < vlen / sizeof(int); ++d) {
+                dd(val);
+            }
+        };
+
+        align(64);
+        L(l_table_constant);
+        broadcast_d(0x3f000000);   // 0.5f
+        dw(0x0001);
+    }
+
+    void uni_vmovshdup(const Xbyak::Xmm &x, const Xbyak::Xmm &y) {
```

Comment:
As far as I can see oneDNN jit_generator already has uni_vmovshdup/uni_vmovhlps mnemonics. Why do we need to overload?

Reply:
This two mnemonics in oneDNN  are not merged since last rebase. Now removed and use oneDNN version. Thank you Dmitry.